src/semanticlayertools/clustering/reports.py

```diff
@@ -0,0 +1,288 @@
+    :type outpath: str
+
+    :param textcolumn: The dataframe column of metadata containing textutal for topic modelling (default=title)
+    :type textcolumn: str
+    :param numberProc: Number of CPU the routine will use (default = all!)
+    :type numberProc: int
+    :param minClusterSize: The minimal cluster size, above which clusters are considered (default=1000)
+    :type minClusterSize: int
+    :param timerange: Time range to evalute clusters for (usefull for limiting computation time, default = (1945, 2005))
+    :type timerange: tuple
+    """
+
+    def __init__(
+        self, infile: str, metadatapath: str, outpath: str,
+        textcolumn: str = 'title',
+        authorColumnName: str = 'author',
+        affiliationColumnName: str = 'aff',
+        publicationIDcolumn: str = 'nodeID',
+        numberProc: int = num_processes,
+        minClusterSize: int = 1000,
+        timerange: tuple = (1945, 2005)
+    ):
+        """Constructor method"""
+        self.numberProc = numberProc
+        self.minClusterSize = minClusterSize
+        self.metadatapath = metadatapath
+        self.textcolumn = textcolumn
+        self.authorColumnName = authorColumnName
+        self.affiliationColumnName = affiliationColumnName
+        self.publicationIDcolumn = publicationIDcolumn
+        clusterdf = pd.read_csv(infile)
+        basedata = clusterdf.groupby(['year', 'cluster']).size().to_frame('counts').reset_index()
+        self.largeClusterList = list(
+            basedata.groupby('cluster').sum().query(f'counts > {self.minClusterSize}').index
+        )
+        self.clusternodes = clusterdf.query(
+            'cluster in @self.largeClusterList'
+        )
+        outfolder = infile.split(os.path.sep)[-1][:-4]
+        self.timerange = timerange
+        self.outpath = os.path.join(outpath, outfolder)
+        if os.path.isdir(self.outpath):
+            raise OSError(f'Output folder {self.outpath} exists. Aborting.')
+        else:
+            os.mkdir(self.outpath)
+            for clu in self.largeClusterList:
+                os.mkdir(os.path.join(self.outpath, f'Cluster_{clu}'))
+
+    def create_corpus(self, dataframe):
+        """Create corpus out of dataframe.
+
+        Using the text contained in the cluster metadata to generate a corpus.
+        After some basic preprocessing each text is used to generate a Spacy doc,
+        of which only the lemmatized words without stop words are considered.
+
+        :params dataframe: Input dataframe
+        :type dataframe: `pd.Dataframe`
+        :returns: A textacy corpus file with english as the base language
+        :rtype: `textacy.Corpus`
+        """
+        mainLanguageCorp = 'en_core_web_lg'
+        nlp = spacy.load(mainLanguageCorp)
+
+        docs = []
+        titles = dataframe[self.textcolumn].values
+        for title in tqdm(titles, leave=False):
+            try:
+                # text pre-processing
```

Comment:
A short comment on the pre-preprocessing could be useful - seems like you're removing some punctuation, some escape symbols and whitespace, but I'm not sure. You're also removing digits?